reverse_file.py

```diff
@@ -1 +1,53 @@
+import os
 
+
+def file_length(name: str) -> int:
+    length = os.stat(name).st_size
+    return length
+
+
+def read_file(name: str, offset: int, size: int, output_buf_offset: int) -> bytearray:
+    """
+    Reads `size` bytes from file `name` starting from `offset` index and
+    stores them into `output_buf` starting from `output_buf_offset`
+  """
+    with open(name, "rb") as f:
+        f.seek(offset)
+        chunk = f.read(size)
+        output_buf = bytearray(chunk[output_buf_offset:])
+        return output_buf
+
+
+def write_file(buf: bytearray, buf_offset: int, size: int, output_file: str, output_file_offset: int) -> None:
+    """
+    Write `size` bytes from `buf` buffer starting from `buf_offset` into `output_file` starting from `output_file_offset`
+  """
+    with open(output_file, "r+b") as f:
+        f.seek(output_file_offset)
+        f.write(bytes(buf[buf_offset:size]))
```

Comment:
There is a bug. Can you find it?

Reply:
"r+" mode only allows reading and writing at the current position in the file, and it does not support seeking or moving the file pointer

Reply:
The bug I meant is in `buf[buf_offset:size]`. The second parameter of slice is end position, not size. So it should be `buf[buf_offset:buf_offset + size]`

Reply:
According to https://stackoverflow.com/a/509014, `r+b` should be fine

Reply:
How could I skip such an obvious bug :/
thanks!